### DISCUSSION TITLE
trying to get https enabled with haproxy in pfsense

### DISCUSSION THREAD

**Discussion by terrorbyte704** [dani-garcia/vaultwarden#4547] [Q&A] [answered]
I just recently installed vaultwarden as a docker, and i wanted to set it up on reverse proxy, but its only comes in http instead of https and I tried to do as the proxy examples but got completely lost as I am using pfsense, but would always gte error 503  Service Unavailable ,any help would be much appreciated,
thanks
![Screenshot 2024-05-08 143054](https://github.com/dani-garcia/vaultwarden/assets/128277877/b77229b5-22d5-43b6-8826-e8ad81bf7d10)

**Comment by devinslick** [comment]
> I just recently installed vaultwarden as a docker, and i wanted to set it up on reverse proxy, but its only comes in http instead of https and I tried to do as the proxy examples but got completely lost as I am using pfsense, but would always gte error 503  Service Unavailable ,any help would be much appreciated,
> thanks
> ![Screenshot 2024-05-08 143054](https://github.com/dani-garcia/vaultwarden/assets/128277877/b77229b5-22d5-43b6-8826-e8ad81bf7d10)
> 

Are you able to get to your vaultwarden instance on http://iPaddress:8080?
Can you share your HAProxy config from pfsense?

**Reply by terrorbyte704** [reply]
Yes i am able to access it,

here is my config

frontend (I do have Use "forwardfor" option enabled)
![Screenshot 2024-05-08 154933](https://github.com/dani-garcia/vaultwarden/assets/128277877/763a0cae-eb61-4b03-8c1d-15db331b8566)

![Screenshot 2024-05-08 154349](https://github.com/dani-garcia/vaultwarden/assets/128277877/04d66aec-c836-403f-a0b5-97876a4fbc9c)

backend

![Screenshot 2024-05-08 155301](https://github.com/dani-garcia/vaultwarden/assets/128277877/c2a052cb-5ab1-4898-9a57-9c0228d8bd43)


![Screenshot 2024-05-08 154849](https://github.com/dani-garcia/vaultwarden/assets/128277877/fd24d83f-e1e7-4cb7-af23-25b6cfe9d89f)

**Comment by terrorbyte704** [comment, marked as answer]
Since no one is helpful enough to help me in the offical discord I will leave my solutions here to help others

first make a front end for port 80, then scroll all the way down to "Default backend, access control lists and actions" then select then create a new action, after that choose use backend and change it to "http-request redirect", in the empty box put "scheme https"

then make a new front end for port 443 and fill out the info like other front end and back proxy, 1 last point in the back don't tick ssl

that it.

**Comment by devinslick** [comment]
@terrorbyte704 , glad you found an answer.  
Nothing struck me as obviously wrong from your screenshots.  Thanks for sharing the details on your config. 
Maybe it should be added to the wiki for a PFSense HAConfig....
https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

**Reply by terrorbyte704** [reply]
glad to have helped
